code/modules/surgery/surgery.dm

```diff
@@ -94,17 +94,19 @@ proc/do_surgery(mob/living/carbon/M, mob/living/user, obj/item/tool)
 				if(step_is_valid == 2) // This is a failure that already has a message for failing.
 					return 1
 				M.op_stage.in_progress += zone
-				S.begin_step(user, M, zone, tool)
 				user.operating = 1//start on it
+				user.fail_next_op = 0
+				S.begin_step(user, M, zone, tool)
 				//We had proper tools! (or RNG smiled.) and user did not move or change hands.
-				if(prob(S.tool_quality(tool)) &&  do_mob(user, M, rand(S.min_duration, S.max_duration)))
+				if(prob(S.tool_quality(tool)) &&  do_mob(user, M, rand(S.min_duration, S.max_duration)) && ((user.mind.isSurgeon) || (!user.mind.isSurgeon) && (prob(50))) && (!user.fail_next_op)) //if they aren't a surgeon, give them a 50% chance of succeeding
```

Comment:
That !user.mind.isSurgeon is completely redundant. (A || !A && B) is equals to just A || B.